python/packages/streamsx/rest_primitives.py

```diff
@@ -49,6 +50,21 @@ def __init__(self, json_rep, rest_client):
 
     def __str__(self):
         return pformat(self.__dict__)
+    __repr__ = __str__
+
+    def refresh(self):
+        """Make a REST call to refresh the JSON object stored within the object
+
+        Returns:
+            _ResourceElement: The provided object instance is returned with updated value
+        """
+        rest_client = self.rest_client
+        rest_self = self.rest_self
+        json_rep = rest_client.make_request(rest_self)
+        self.__dict__.clear()
```

Comment:
>  The design makes it difficult to clear these attributes on subsequent refresh(), because I don't keep track of the keys that are added to the object.

Not sure why this is an issue. The reality is the same set of attributes are returned every time.